### DISCUSSION TITLE
Omit DateTime scalar from built schema

### DISCUSSION THREAD

**Discussion by robinelvin** [MichalLytek/type-graphql#1333] [Q&A]
I am using TypeGraphQL to generate schema which is being passed to Neo4jGraphQL. Since both libraries define a `DateTime` scalar this throws a schema error. My initial workaround was to generate the schema string and remove the scalar definition but the string output omits the directives I need.

Is there a way to omit the DateTime scalar from the `buildSchema` operation?

EDIT:

I am being stupid. From [Emitting the schema SDL](https://typegraphql.com/docs/emit-schema.html) I can see there is a method you can use called `printSchemaWithDirectives` This is now working although it would be nice if the schema did not have to be converted to a string in order to remove the scalar definition.

**Comment by MichalLytek** [comment]
https://typegraphql.com/docs/scalars.html#custom-scalars
```ts
const schema = await buildSchema({
  resolvers,
  scalarsMap: [{ type: Date, scalar: DifferentDateTimeScalar }],
});
```

**Reply by robinelvin** [reply]
I thought this was the answer as this got rid of the naming clash but after deeper inspection this has defined all `DateTime` fields as `DifferentDateTimeScalar`. This means that Neo4jGraphQL does not consider these fields to be `DateTime`.

What I need is to omit the scalar definition from the output. Something like this:

```javascript
const schema = await buildSchema({
  resolvers,
  omittedTypes: ['DateTime'],
});
```

I simply want to define a field as a `DateTime` and let Neo4jGraphQL add the scalars to the schema. See [Neo4jGraphQL DateTime](https://neo4j.com/docs/graphql-manual/current/type-definitions/types/#_datetime) for my use case.

**Reply by MichalLytek** [reply]
I think TypeGraphQL only emits the used scalars.

> This means that Neo4jGraphQL does not consider these fields to be DateTime.

In graphql-js you can't just set a field to some scalar name without defining the scalar (aka scalar resolver).
So `DifferentDateTimeScalar` is the one that Neo4J uses, so that when you query your schema, the fields will have the same format/return logic.

**Reply by robinelvin** [reply]
> In graphql-js you can't just set a field to some scalar name without defining the scalar (aka scalar resolver). So `DifferentDateTimeScalar` is the one that Neo4J uses, so that when you query your schema, the fields will have the same format/return logic.

The point I'm trying to make is that Neo4jGraphQL defines the scalar by augmenting the schema. My workaround is to emit the schema to a string, remove the `scalar DateTime` line which TypeGraphQL creates and then use in Neo4jGraphQL but this seems a bit wasteful.

e.g.

```typescript
const neoSchema = new Neo4jGraphQL({
    config: {
        enableDebug: true,
    },
    typeDefs: typeDefs.replace('scalar DateTime', ''), // Remove scalar added by TypeGraphQL
    driver,
});
```

**Reply by MichalLytek** [reply]
I'm unable to change anything in order to not emit the scalar - it's out of my control. You need to keep removing it from typedefs string by yourself.
